test/e2e/os.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kubermatic/kubeone/test/e2e/provisioner"
+)
+
+type OperatingSystem string
+
+const (
+	OperatingSystemUbuntu  OperatingSystem = "ubuntu"
+	OperatingSystemCentOS  OperatingSystem = "centos"
+	OperatingSystemCoreOS  OperatingSystem = "coreos"
+	OperatingSystemDefault OperatingSystem = ""
+)
+
+func ValidateOperatingSystem(osName string) error {
+	switch OperatingSystem(osName) {
```

Comment:
in Go switch doesn't do implicit fallthrough, this code will result an error even with valid os choices.

```suggestion

	switch OperatingSystem(osName) {
	case OperatingSystemUbuntu, OperatingSystemCentOS, OperatingSystemCoreOS, OperatingSystemDefault:
		return nil
	}
	return errors.New("failed to validate operating system")
```